database/migrations/2024_08_04_135449_create_student_associations_table.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+use App\Models\Edition;
+use App\Models\StudentAssociation;
+use App\Models\User;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+use function DatabaseHelpers\createManyToManyRelation;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('student_associations', function (Blueprint $table) {
+            $table->id();
+            $table->foreignIdFor(User::class)->unique()->constrained()->cascadeOnDelete();
+            $table->timestamps();
+            $table->string('name');
+            $table->string('code');
+            $table->integer('points')->unsigned()->default(0);
```

Comment:
I added this as the points column was not available with the createManyToManyRelation below.

Reply:
what do you mean exactly by this?

Reply:
If you are refering to the fact that we can't programatically access the points, take a look at [this documentation page]([s4443330@vuw.leidenuniv.nl](mailto:s4443330@vuw.leidenuniv.nl)). We already use this actually, under https://github.com/NIAEFEUP/enei-website/blob/main/app/Models/EnrollmentProduct.php

Reply:
Reverted this back to the original code, left your changes commented out. If we have problems we can just uncomment them again